pkg/controllers/data.go

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	. "github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/interactor"
+	"github.com/gorilla/mux"
+)
+
+type DataController struct {
+	DataInteractor interactor.DataInteractor
+}
+
+type DataInterface interface {
+	GetAll()
+	GetByID()
+	Create()
+}
+
+type errorMessage struct {
+	error   bool
+	message string
+}
+
+func NewDataController(dataInteractor interactor.DataInteractor) DataController {
+	return DataController{dataInteractor}
+}
```

Comment:
In order to avoid allocations, as a rule of thumb, [always return a pointer](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) for constructors (I'm not quite convinced for other cases, but for constructors I'm quite sure, since they should only run once).

In conclusion, this (and all other constructors) should return a pointer :eyes: 
```suggestion
func NewDataController(dataInteractor interactor.DataInteractor) *DataController {
	return &DataController{dataInteractor}
}
```

Reply:
Done!